src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -87,10 +88,18 @@ internal abstract class BrokerClient : IBrokerClient {
         protected abstract void OnCredentialsValidated(bool isValid);
 
         public async Task PingAsync() {
-            try {
-                (await HttpClient.PostAsync("/ping", new StringContent(""))).EnsureSuccessStatusCode();
-            } catch (HttpRequestException ex) {
-                throw new RHostDisconnectedException(Resources.Error_HostNotResponding.FormatInvariant(ex.Message), ex);
+            if (HttpClient != null) {
+                var isOnline = await CheckMachineOnlineAsync();
```

Comment:
`ConnectAsync` is called not only when user switches session, but also when he tries to reconnect to the current one

---
In reply to: [81587164](https://github.com/Microsoft/RTVS/pull/2416#discussion_r81587164) [](ancestors = 81587164)